zone/gm_commands/worldwide.cpp

```diff
@@ -2,20 +2,57 @@
 
 void command_worldwide(Client *c, const Seperator *sep)
 {
-	std::string sub_command;
-	if (sep->arg[1]) {
-		sub_command = sep->arg[1];
+	int arguments = sep->argnum;
+	if (!arguments) {
+		c->Message(Chat::White, "Usage: #worldwide cast [Spell ID] [Disable Message] - Cast a spell worldwide, use 1 for Disable Message to not send a global message");
+		c->Message(Chat::White, "Usage: #worldwide message [Message] - Send a message worldwide");
+		c->Message(Chat::White, "Usage: #worldwide move [Zone ID|Zone Short Name] - Move all players worldwide to specified zone");
+		c->Message(Chat::White, "Usage: #worldwide moveinstance [Instance ID] - Move all players worldwide to specified instance");
+		c->Message(Chat::White, "Usage: #worldwide remove [Spell ID] - Remove a spell worldwide");
+		return;
 	}
+	
+	bool is_cast = !strcasecmp(sep->arg[1], "cast");
+	bool is_remove = !strcasecmp(sep->arg[1], "remove");
+	bool is_message = !strcasecmp(sep->arg[1], "message");
+	bool is_move = !strcasecmp(sep->arg[1], "move");
+	bool is_moveinstance = !strcasecmp(sep->arg[1], "moveinstance");
 
-	if (sub_command == "cast") {
-		if (sep->arg[2] && Seperator::IsNumber(sep->arg[2])) {
-			uint8 update_type     = WWSpellUpdateType_Cast;
-			auto  spell_id        = std::stoul(sep->arg[2]);
-			bool  disable_message = false;
-			if (sep->arg[3] && Seperator::IsNumber(sep->arg[3])) {
+	if (
+		!is_cast &&
+		!is_message &&
+		!is_move &&
+		!is_moveinstance &&
+		!is_remove
+	) {
+		c->Message(Chat::White, "Usage: #worldwide cast [Spell ID] [Disable Message] - Cast a spell worldwide, use 1 for Disable Message to not send a global message");
+		c->Message(Chat::White, "Usage: #worldwide message [Message] - Send a message worldwide");
+		c->Message(Chat::White, "Usage: #worldwide move [Zone ID|Zone Short Name] - Move all players worldwide to specified zone");
+		c->Message(Chat::White, "Usage: #worldwide moveinstance [Instance ID] - Move all players worldwide to specified instance");
+		c->Message(Chat::White, "Usage: #worldwide remove [Spell ID] - Remove a spell worldwide");
+		return;
+	}
+
+	if (is_cast) {
+		if (sep->IsNumber(2)) {
```

Comment:
Should these be arg[2] ? A couple of those in here.

Reply:
The method checks if sep->arg[2] is a number.